lib/compat/wordpress-5.9/class-gutenberg-rest-navigation-controller.php

```diff
@@ -0,0 +1,176 @@
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		// Lists/updates a single nav item based on the given id.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\/\w-]+)',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'id' => array(
+							'description' => __( 'The id of a navigation', 'gutenberg' ),
+							'type'        => 'string',
+						),
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+					'args'                => array(
+						'force' => array(
+							'type'        => 'boolean',
+							'default'     => false,
+							'description' => __( 'Whether to bypass Trash and force deletion.', 'gutenberg' ),
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks if the user has permissions to make the request.
+	 *
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	protected function permissions_check() {
+		// Verify if the current user has edit_theme_options capability.
+		// This capability is required to edit/view/delete templates.
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_cannot_manage_templates',
+				__( 'Sorry, you are not allowed to access the templates on this site.', 'gutenberg' ),
+				array(
+					'status' => rest_authorization_required_code(),
+				)
+			);
+		}
+
+		return true;
+	}
+
+	public function filter_response_by_context( $data, $context ) {
+		$data['id'] = $data['slug'];
+		return $data;
+	}
+
+	private function prepare_request( $request ) {
+		return $request;
+		$slug = $request['id'];
+		$wp_query_args        = array(
+			'post_name__in'  => array( $slug ),
+			'post_type'      => 'wp_navigation',
+			'post_status'    => array( 'auto-draft', 'draft', 'publish', 'trash' ),
+			'posts_per_page' => 1,
+			'no_found_rows'  => true,
+		);
+		$template_query       = new WP_Query( $wp_query_args );
+		$post_id = $template_query->posts[0]->ID;
+
+		$request['slug'] = $request['id'];
```

Comment:
```suggestion
		$request['slug'] = $request['slug'];
```

Reply:
or was that intentional?

Reply:
This code doesn't seem to get executed because @adamziel returns the value here: https://github.com/WordPress/gutenberg/pull/36522/files#diff-678d8bd4bed4738f7b7c74ec8f9eb0bfdcc45d87ae275466cce7cb3601e654ecR125
I suppose that is intentional, @adamziel ?

Reply:
It is, that's a very quick&dirty draft and I just wanted to get some data flowing